### DISCUSSION TITLE
Get color Tilelayer

### DISCUSSION THREAD

**Discussion by Carlos-Carreno-Berlanga** [visgl/deck.gl#6938] [Q&A]
Hi,

I am using TileLayer to display raster data. It works fine, but I would like to display tooltip information depending on the color where the cursor mouse is hovering.
Unfortunately is not working because the color information is not equal to the png files of the raster data.

**Comment by Pessimistress** [comment]
https://deck.gl/docs/api-reference/layers/bitmap-layer#pixel-picking

**Reply by Carlos-Carreno-Berlanga** [reply]
Hi Thanks for your help,

I tried that but unfortunately doesn't work. The click event doesn't fire for sublayers. And I haven't been able to make it work directly on TileLayer. There is a color property on the pickingInfo but it doesn't match with the real color of the raster layer. My use case is very simple, is a weathermap so basically if the user clicks or hovers the mouse, we should tell them the temperature based on the color.

**Reply by Pessimistress** [reply]
> I tried that but unfortunately doesn't work.

Can't help you unless you share your code. You can see the coordinates by adding the following to the tileLayer in [this example](https://github.com/visgl/deck.gl/blob/master/examples/website/map-tile/app.js#L41):

`onHover: info => console.log(info.bitmap)`

`pickingInfo.color` is an encoded pixel from the picking buffer. It has nothing to do with the pixel color on screen.

**Reply by Carlos-Carreno-Berlanga** [reply]
Hi,

Thanks a lot for the interest. I created a codepen explaining the issue, https://codepen.io/xasif14964-dufeed-com/pen/poawVEo?editors=001 , basically I would like to know the color when the user hovers the mouse over the TileLayer.

**Reply by Pessimistress** [reply]
I don't see where you tried the color query approach from the BitmapLayer documentation.

**Reply by Carlos-Carreno-Berlanga** [reply]
Sorry I modified the code, including the approach that you mention.
https://codesandbox.io/s/deck-gl-di-tu-shang-nibideowobiao-shi-suru-forked-jhp5g6?file=/src/App.js:0-2656

**Reply by Pessimistress** [reply]
When you listen to `onClick` of the TileLayer, `pickInfo.layer` is the TileLayer, and `pickInfo.sourceLayer` is the BitmapLayer where the event originates from.

```js
onClick: (pickInfo) => {
  console.log(pickInfo); // observe the event here
  const { bitmap, sourceLayer } = pickInfo;
  // continue to access sourceLayer.props.image
}
```

**Reply by Carlos-Carreno-Berlanga** [reply]
Thanks a lot, I spent at least 2 days trying to figure it out how to do this. I created another sandbox just in case someone else has the same issue as me.
https://codesandbox.io/s/deck-gl-di-tu-shang-nibideowobiao-shi-suru-forked-23k1dd
